lib/domain/entities/track_segment.dart

```diff
@@ -0,0 +1,15 @@
+import 'package:freezed_annotation/freezed_annotation.dart';
+import 'package:movna/domain/entities/track_point.dart';
+
+part 'track_segment.freezed.dart';
+
+/// Continuous track segment followed by user, without any pause.
+@freezed
+class TrackSegment with _$TrackSegment {
+  const TrackSegment._();
+
+  const factory TrackSegment({
+    /// All track points followed by user.
+    @Default([]) List<TrackPoint> trackPoints,
```

Comment:
I know we can always find that value using trackpoints but I think a startDate and endDate would make sense here

also, would an automatic pause create multiple segments or would just manual pause create them ?

Reply:
Yes I am not sure a startTime and stopTime field would be relevant here since we can get these values from the Trackpoints, wouldn't getters suffice ?

Reply:
An automatic pause would behave the same as a manuel pause there. The aim of the intermediate class between `Activity` and `TrackPoint` is to be able to know where and when pauses happened, to be able to draw different colors on the map.

|======================-----------------------------=====================|
start                                             pause                                   resume                                        stop

Reply:
> Yes I am not sure a startTime and stopTime field would be relevant here since we can get these values from the Trackpoints, wouldn't getters suffice ?

Getters would suffice I guess yes